components/button/floating-buttons.js

```diff
@@ -0,0 +1,236 @@
+import 'lit-media-query/lit-media-query.js';
+import { css, html, LitElement } from 'lit-element/lit-element.js';
+import { ButtonMixin } from './button-mixin.js';
+import { buttonStyles } from './button-styles.js';
+import { labelStyles } from '../typography/styles.js';
+
+class FloatingButtons extends ButtonMixin(LitElement) {
+	static get properties() {
+		return {
+			/**
+			 * Whether always enable floating buttons. Note: by default, buttons may not float depending on
+			 * space available since floating buttons may result in poor user experience when view-port is small
+			 * (ex. phones).
+			 */
+			alwaysFloat: {
+				reflect: true,
+				type: Boolean,
+				attribute: 'always-float'
+			},
+
+			/**
+			 * Minimum height of view-port in order for buttons to float.
+			 */
+			minHeight: {
+				reflect: true,
+				type: String,
+				attribute: 'min-height'
+			},
+
+			_viewportIsAtLeastMinHeight: {
+				reflect: true,
+				type: Boolean
+			}
+		};
+	}
+
+	static get styles() {
+		return [ labelStyles, buttonStyles,
+			css`
+			:host {
+				box-sizing: border-box;
+				display: block;
+			}
+			:host([hidden]) {
+				display: none;
+			}
+			.d2l-floating-buttons-container {
+				border-top: 1px solid transparent;
+				display: block;
+				margin: 0 auto;
+				width: 100%;
+			}
+			.d2l-floating-buttons-container.d2l-floating-buttons-floating {
+				animation: d2l-floating-buttons-animation 500ms ease-out;
+				-webkit-animation: d2l-floating-buttons-animation 500ms ease-out;
+				background-color: #ffffff;
+				background-color: rgba(255, 255, 255, 0.88);
+				border-top-color: var(--d2l-color-mica);
+				bottom: 0;
+				box-shadow: 0 -2px 4px rgba(86, 90, 92, .2);
+				left: 0;
+				position: fixed;
+				right: 0;
+				z-index: 999;
+			}
+			.d2l-floating-buttons-container > div {
+				padding: 0.75rem 0 0 0;
+				position: relative;
+			}
+			.d2l-floating-buttons-container > div {
+				padding: 0.75rem 0 0 0;
+				position: relative;
+			}
+			.d2l-floating-buttons-inner-container ::slotted(d2l-button),
+			.d2l-floating-buttons-inner-container ::slotted(button),
+			.d2l-floating-buttons-inner-container ::slotted(.d2l-button) {
+				margin-right: var(--d2l-button-spacing) !important;
```

Comment:
`--d2l-button-spacing` doesn't appear to be defined. In polymer 3 button, it was defined in the shared styles. From what I can tell, it doesn't seem like it actually needs to be a var

Reply:
I found it in the old version `0.75rem` is there something I can get to scan for undefined/unused css variables? might be handy to make that part of my pre-commit check list.

Reply:
Not that I am aware of.  I think we're mostly familiar with what's going on here, so we "just know".  My suggestion is to take your time, and scrutinize anything that isn't clear.  :)